locallibrary/catalog/views.py

```diff
@@ -1,3 +1,27 @@
 from django.shortcuts import render
+from catalog.models import Book, Author, BookInstance, Genre
 
-# Create your views here.
+
+def index(request):
+    """View function for home page of site."""
+
+    # Generate counts of some of the main objects
+    num_books = Book.objects.count()
+    num_instances = BookInstance.objects.all().count()
+
+    # Available books (status = 'a')
+    num_instances_available = BookInstance.objects.filter(
+        status__exact='a').count()
```

Comment:
`__exact` ngoài cái này thì còn những field lookup nào khác em nhỉ?

Reply:
## Ngoài exact ra thì còn những field look up khác:
- Các field liên quan tới số:
  - lt: Nhỏ hơn giá trị cho trước
  - lte: Nhỏ hơn hoặc bằng giá trị cho trước
  - gt: Lớn hơn giá trị cho trước
  - gte: Lớn hơn hoặc bằng giá trị cho trước
  - range: Trong một khoảng nhất định
 

- Các field liên quan tới ngày tháng:
  - month: Trả về các giá trị trong tháng cho trước
  - day: Trả về các giá trị trong ngày cho trước
  - week: Trả về các giá trị trong tuần cho trước
  - year: Trả về các giá trị trong năm cho trước
  - hour: Trả về các giá tị trong giờ cho trước
  - minute: Trả về các giá tị trong phút cho trước
  - second: Trả về các giá trị trong giây cho trước


- Các field liên quan tới text
  - contains: Trả về giá trị chứa đoạn text cho trước
  - icontains: Giống trên nhưng mà ignoring case
  - startswith: Trả về giá trị bắt đầu bằng giá trị cho trước
  - endswith: Trả về giá trị kết thúc bằng giá trị cho trước
  - regex: Giá trị phải match regex expression
  - iregex: Giống trên nhưng mà ignoring case.
  - in: Trả lại giá trị có trong tuple, set, list